Source/WebKit/WebProcess/WebPage/WebFrame.cpp

```diff
@@ -1113,4 +1113,11 @@ uint64_t WebFrame::messageSenderDestinationID() const
     return m_frameID.object().toUInt64();
 }
 
+OptionSet<WebCore::NetworkConnectionIntegrity> WebFrame::networkConnectionIntegrityPolicy() const
+{
+    if (!coreFrame() || !coreFrame()->document() || !coreFrame()->document()->topDocument().loader())
```

Comment:
auto* coreFrame = this->coreFrame();
if (!coreFrame)
    return { };
auto* document = coreFrame->document();
if (!document)
    return { };
auto* topDocumentLoader = document->topDocument().loader();
if (!topDocumentLoader)
    return { };
return topDocumentLoader->networkConnectionIntegrityPolicy();

That way, we don't have any duplicate function calls.

Also, I'm not sure how this interacts with https://github.com/WebKit/WebKit/pull/13108 where we're no longer always using the top document loader as the source.  One of them needs to be updated, I think.

Reply:
Yes, this function will need to change. I'll land this patch first and change this function in the other patch to check the current frame's document loader whenever the main frame URL has a custom scheme.